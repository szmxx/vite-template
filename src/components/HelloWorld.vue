<!--
 * @Author: cola
 * @Date: 2022-07-04 16:13:38
 * @LastEditors: cola
 * @Description:
-->
<template>
  <div>{{ msg }}</div>
  <div>
    <button @click="greet">greet</button>
  </div>
</template>

<script setup lang="ts">
  import useStore from '@/store/demo'
  const store = useStore()
  const { msg } = storeToRefs(store)
  const greet = () => {
    msg.value = 'greeting'
    store.greet()
  }
</script>

<style scoped lang="scss">
  div {
    color: $black;
  }
</style>
