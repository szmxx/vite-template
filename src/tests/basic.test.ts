/*
 * @Author: cola
 * @Date: 2022-07-07 14:34:41
 * @LastEditors: cola
 * @Description:
 */
import { mount } from '@vue/test-utils'
import HelloWorld from '@/components/HelloWorld.vue'
test('mount component', async () => {
  expect(HelloWorld).toBeTruthy()

  const wrapper = mount(HelloWorld, {})
  expect(wrapper.text()).toContain('hello world')
  expect(wrapper.html()).toMatchSnapshot()

  await wrapper.get('button').trigger('click')
  expect(wrapper.text()).toContain('greeting')
})
