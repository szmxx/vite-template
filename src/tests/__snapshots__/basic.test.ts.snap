exports[`mount component 1`] = `
"<div>hello world</div>
<div><button> greet </button></div>"
`;
